### ISSUE TITLE
SSH connection to proxmox nodes doesn't use ssh/config 

### ISSUE THREAD

**Issue by bitchecker** [bpg/terraform-provider-proxmox#812]
**Describe the bug**
in the provider section it is possible to configure all the nodes you need to reach (single node or cluster) but the only authentication method that is working is using password.

**To Reproduce**
disable password authentication and it will be impossible to connect to the node

Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl
provider "proxmox" {
  endpoint = "https://${var.pm_host}:${var.pm_webport}/"
  username = "${var.pm_user}@pam"
  password = var.pm_password
  insecure = true
  tmp_dir  = "${path.module}"
  ssh {
    agent = true
    dynamic "node" {
      for_each = var.pm_nodes
      content {
        name    = node.value.name
        address = node.value.address
        port    = node.value.port
      }
    }
  }
}
```

**Expected behavior**
ssh connection running well using the `.ssh/config` specs and authentication method too

- Single or clustered Proxmox: `all`
- Provider version (ideally it should be the latest version): `latest`
- Terraform version: `v1.6.6`
- OS (where you run Terraform from): `Fedora 39`

**Comment by bpg** [comment]
Hi @bitchecker!

>in the provider section it is possible to configure all the nodes you need to reach (single node or cluster) but the only authentication method that is working is using password.

I noticed in your config that the SSH agent is enabled (`agent = true`), so authentication via the agent is set as the primary method, while password authentication serves as a fallback. The provider itself doesn't directly use configuration from `.ssh/config`, but `ssh-agent` does. Utilizing the agent allows you to apply any node-specific SSH configuration options.

If your SSH agent isn't running or doesn't have the required keys loaded, it might be causing the issue. For troubleshooting common agent-related issues, you can refer to a helpful summary [here](https://github.com/bpg/terraform-provider-proxmox/issues/619#issuecomment-1760582479).

**Comment by bitchecker** [comment]
Hi @bpg,
I tried to disable again password authentication and configure `ssh-agent` as suggested, it seems that it works fine.
Thanks a lot for this!

I have another question: the nodes section doesn't accept hostnames but only IPs, so every time I need to check if the public IP is changed in the meantime.

There is a specific reason to disable hostnames in nodes configuration?

**Comment by bpg** [comment]
There is no reason, I just didn't think about this use case 😅.

Originally, the list of servers in the `ssh` block was added to help obtain IP addresses for PVE nodes that don't have static IPs.

When the provider establishes an SSH connection to a node, it first calls the PVE API to list the node interfaces and selects the first non-empty IP from the list. In typical home lab deployments, the node network interfaces look like this, making IP lookup trivial:
![Screenshot](https://github.com/bpg/terraform-provider-proxmox/assets/627562/282436bd-f8f9-4df8-849a-8e413bd2be07)

However, there are cases where the network setup is more complicated, with dedicated management networks, DHCP addresses, etc. Therefore, we added support for the `ssh.node` block to override the IP resolution with known node addresses.

I think it makes sense to just remove the [validator](https://github.com/bpg/terraform-provider-proxmox/blob/main/proxmoxtf/provider/schema.go#L158) and allow any hostname there.

Also, if your PVE node names are resolvable, you may benefit from the proper [DNS lookup](https://github.com/bpg/terraform-provider-proxmox/pull/725/commits/7a7693558c1f040d40837facb1826167a97f4520#diff-376eaed8b58fe185cd4216c58b672426603406af9fcdfe98c20b470104a76775R232) proposed in #725, though it seems the author has abandoned the PR. However, I'm considering re-implementing this feature separately...

**Comment by bpg** [comment]
Hey @bitchecker 👋🏼 

Are you up for another PR to close this ticket? It involves replacing `validation.IsIPAddress` with `validation.StringIsNotEmpty` [here](https://github.com/bpg/terraform-provider-proxmox/blob/main/proxmoxtf/provider/schema.go#L158). The change is straightforward, and I hope you'll be able to test the modified provider in your environment, and also provide some feedback if [CONTRIBUTING.md](https://github.com/bpg/terraform-provider-proxmox/blob/main/CONTRIBUTING.md) has enough details to get you up and running.

Thanks!

**Comment by bitchecker** [comment]
> The provider itself doesn't directly use configuration from `.ssh/config`, but `ssh-agent` does. Utilizing the agent allows you to apply any node-specific SSH configuration options.

Hi @bpg,
while I'm waiting for new release, I was trying to use ssh connection via bastion host (is another use case) and I can see that also using the `ssh-agent` the configurations are not loaded.

I think that we need something that uses the system ssh.

**Comment by bpg** [comment]
The ssh connection via bastion will be addressed in #852.
